coconut/src/proof/messages_pok/mod.rs

```diff
@@ -558,14 +558,23 @@ mod tests {
     #[test]
     fn empty_proof() {
         let mut rng = StdRng::seed_from_u64(0u64);
-        let (_sk, _pk, params, _messages) = test_setup::<Bls12_381, Blake2b512, _>(&mut rng, 1);
+        let (_, _, params, messages) = test_setup::<Bls12_381, Blake2b512, _>(&mut rng, 1);
         let h = G1::rand(&mut rng).into_affine();
 
-        assert_eq!(
-            MessagesPoKGenerator::init(&mut rng, &[], &params, &h),
-            Err(MessagesPoKError::MessageInputError(
-                MessageUnpackingError::NoMessagesProvided
-            ))
-        );
+        let pok = MessagesPoKGenerator::init(&mut rng, &[], &params, &h)
+            .unwrap();
+
+        let mut chal_bytes_prover = vec![];
```

Comment:
As you are using the same variables for prover's and verifier's challenge, they should not have suffix `_prover`, `chal_bytes` is good.